### PR TITLE
[SPARK-180] Implement history server link in Dispatcher UI

### PR DESCRIPTION
**Pull request by s-urbaniak**

### CODE REVIEW

**Comment by mgummelt** [comment]
lol. plz rebase. bunch of extra commits.

File: core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala [hunk 1]
```diff
@@ -366,10 +373,14 @@ private[spark] class MesosClusterScheduler(
     } else {
       ""
     }
+
     val envBuilder = Environment.newBuilder()
-    desc.command.environment.foreach { case (k, v) =>
+    adjust(desc.command.environment, "SPARK_SUBMIT_OPTS", "")(
```
Comment (mgummelt) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala]: I just spoke with @sttts about this.  Let's use --conf.

**Comment by skonto** [comment]
@s-urbaniak i tried to review it locally, but checking out the PR, building spark and running it against our suite, made our cluster mode tests fail... could you check that pls? All cluster tests gave a time out...
Run starting. Expected test count is: 9
SparkJobSpec:
- simple sum in fine-grained mode
- simple collect in fine-grained mode *** FAILED ***
  The test did not complete within the specified 300 seconds time limit. (TimeLimitedTests.scala:137)
- simple count in coarse-grained mode *** FAILED ***
  org.apache.spark.SparkException: Error starting driver, DRIVER_ABORTED
  at org.apache.spark.scheduler.cluster.mesos.MesosSchedulerUtils$$anon$1.run(MesosSchedulerUtils.scala:122)
  ...
- spark.cores.max property should be honored in coarse grain mode *** FAILED ***
  org.apache.spark.SparkException: Error starting driver, DRIVER_ABORTED
  at org.apache.spark.scheduler.cluster.mesos.MesosSchedulerUtils$$anon$1.run(MesosSchedulerUtils.scala:122)

**Comment by mgummelt** [comment]
Hey @skonto, I got the same error and fixed it in our own private fork.  We're not merging this upstream, so no need to test, but thanks.  In the future, we won't submit private PRs to this repo.

Let me know If you're interested in this functionality being merged upstream.  We're still working out the process for what does and doesn't get contributed.

**Pull request closed without merging**